src/datachain/lib/file.py

```diff
@@ -190,6 +190,21 @@ def __init__(self, **kwargs):
         self._catalog = None
         self._caching_enabled: bool = False
 
+    @classmethod
+    def upload(
+        cls, path: str, data: bytes, catalog: Optional["Catalog"] = None
+    ) -> "File":
+        if catalog is None:
```

Comment:
[F] main problems here are:

- we have to instantiate a new catalog for each upload
- same but for client

Reply:
That's okay, I think. This is what we are doing everywhere (check the `file.open()` for example).

Besides, initializing `Client` should be cheap - fsspec's filesystem instances get cached automatically.

This is what bothers me the most at the moment though, would be nice if these things were done lazily and only once.

https://github.com/iterative/datachain/blob/d0c22fc0fa28112bf0b6e51b8757e7602247c269/src/datachain/client/s3.py#L50-L56